### ISSUE TITLE
BatchLabs hangs after selecting Local Template

### ISSUE THREAD

**Issue by whatsondoc** [Azure/BatchLabs#1322]
**I'm submitting a ...**  (check one with "x")
[X] bug report
[ ] feature request

**Description**
Running on Mac OS v10.13.4 (17E199). 
Recently installed BatchLabs 0.14.1, opened the app and authenticated. Able to view Pools, Jobs, interact as expected and perform operations (delete pools, for example).
Gallery > Local Template, and select the attached job creation template (inclusive of TaskFactory).
[batch-job-TF-ParametricSweep.azuredeploy.txt](https://github.com/Azure/BatchLabs/files/1965656/batch-job-TF-ParametricSweep.azuredeploy.txt)

The json is validated (and shows as valid), however after this BatchLabs becomes unresponsive. Nothing is shown afterwards, and clicking through any other icons produces no result (aside from a couple of options on the Profile, 'About', 'Report a bug...', 'View Logs').

**Instruction to repro the problem if applicable**

**Version of BatchLabs(For bugs)**
Version: 0.14.1

**Logs, for bugs(Click profile > View Logs > app.log, client.log)**
[BatchLabs_logs.zip](https://github.com/Azure/BatchLabs/files/1965629/BatchLabs_logs.zip)

**Comment by ascobie** [comment]
Thanks for reporting this. Will try and replicate the issue shortly with your template.

________________________________

**Comment by ascobie** [comment]
Hi There, 
I have identified the issue. This was a bug in 14.1 that has since been fixed but not released as yet. Tim is working on the 15.0 release at the moment so it will be out shortly. In the mean time, I have attached the template with a workaround that fixes the issue. 

[batch-job-TF-ParametricSweep.azuredeploy.json.txt](https://github.com/Azure/BatchLabs/files/1965743/batch-job-TF-ParametricSweep.azuredeploy.json.txt)

The issue was that it always looks for an advanced data type on the metadata object of the parameter. If there was no metadata object declared the application crashed and got into a bad state. 

Cheers
